### ISSUE TITLE
LightAPI has some issue (with Aurora)

### ISSUE THREAD

**Issue by IPECTER** [BeYkeRYkt/LightAPI#23]
I'm using a server with StarLight.
And I have to use LightAPI 5.0.0 or higher.

And Aurora also uses it.

Because I have to use it with LightAPI, https://github.com/BeYkeRYkt/Aurora
I use this.

But there's an issue with this.

often
"Light" occurs in a place where there's
"Light" disappears when particles haven't disappeared.
"Light" doesn't occur exactly at the location of the particles.

Aurora's developer said
 "I can't update Aurora with new LightAPI compatibility "
"because new LightAPI is not in maven and It is too difficult"

I must use Aurora, which works well with LightAPI.
Can you fix it?
Please....

**Comment by QuintenQVD0** [comment]
bump i have the same issue

**Comment by OpticFusion1** [comment]
It seems like quite a bit has changed since I last looked at LightAPI, if @BeYkeRYkt is willing to PR the changes to Aurora needed to bump the used LightAPI version to 5.1.0, then I'll gladly update it :)

**Comment by IPECTER** [comment]
> It seems like quite a bit has changed since I last looked at LightAPI, if @BeYkeRYkt is willing to PR the changes to Aurora needed to bump the used LightAPI version to 5.1.0, then I'll gladly update it :)

WTF he is god he is god he is god

**Comment by OpticFusion1** [comment]
Pushed an update out :p

**Comment by OpticFusion1** [comment]
Turns out, gotta do something with prepare()?
Literally nothing goes into this however documentation or example wise.

**Comment by BeYkeRYkt** [comment]
> Turns out, gotta do something with prepare()? Literally nothing goes into this however documentation or example wise.

prepare() and initialization() are intended for internal use.

**Comment by OpticFusion1** [comment]
> > Turns out, gotta do something with prepare()? Literally nothing goes into this however documentation or example wise.
> 
> prepare() and initialization() are intended for internal use.

Well then, you'll have to look at the current code and see why it's saying to use prepare(), I'm using your forks code.

**Comment by BeYkeRYkt** [comment]
> > > Turns out, gotta do something with prepare()? Literally nothing goes into this however documentation or example wise.
> > 
> > 
> > prepare() and initialization() are intended for internal use.
> 
> Well then, you'll have to look at the current code and see why it's saying to use prepare(), I'm using your forks code.

An example of usage is [here](https://github.com/BeYkeRYkt/LightAPI/blob/5.0/bukkit-common/src/main/java/ru/beykerykt/minecraft/lightapi/bukkit/BukkitPlugin.java#L60). However, you do not need to do this, since LightAPI-Bukkit does it itself when loading and starting the server.

**Comment by OpticFusion1** [comment]
> > > > Turns out, gotta do something with prepare()? Literally nothing goes into this however documentation or example wise.
> > > 
> > > 
> > > prepare() and initialization() are intended for internal use.
> > 
> > 
> > Well then, you'll have to look at the current code and see why it's saying to use prepare(), I'm using your forks code.
> 
> An example of usage is [here](https://github.com/BeYkeRYkt/LightAPI/blob/5.0/bukkit-common/src/main/java/ru/beykerykt/minecraft/lightapi/bukkit/BukkitPlugin.java#L60). However, you do not need to do this, since LightAPI-Bukkit does it itself when loading and starting the server.

Tell that to your library then.

https://paste.md-5.net/duxefaruku.md

**Comment by BeYkeRYkt** [comment]
> > > > > Turns out, gotta do something with prepare()? Literally nothing goes into this however documentation or example wise.
> > > > 
> > > > 
> > > > prepare() and initialization() are intended for internal use.
> > > 
> > > 
> > > Well then, you'll have to look at the current code and see why it's saying to use prepare(), I'm using your forks code.
> > 
> > 
> > An example of usage is [here](https://github.com/BeYkeRYkt/LightAPI/blob/5.0/bukkit-common/src/main/java/ru/beykerykt/minecraft/lightapi/bukkit/BukkitPlugin.java#L60). However, you do not need to do this, since LightAPI-Bukkit does it itself when loading and starting the server.
> 
> Tell that to your library then.
> 
> https://paste.md-5.net/duxefaruku.md

My library doesn't say anything.
![изображение](https://user-images.githubusercontent.com/2690323/150682537-0aa00734-8927-45f2-84a5-fc6d5dea8bf1.png)
What configuration are you running on ?

**Comment by OpticFusion1** [comment]
Aurora [4.1](https://github.com/OpticFusion1/Aurora/commit/1e506b6c14e2f01dd644f7955e63797596488225) and https://www.spigotmc.org/resources/lightapi.4510/ 5.1.0

**Comment by BeYkeRYkt** [comment]
> Aurora 4.1 and https://www.spigotmc.org/resources/lightapi.4510/ 5.1.0

Paper ?

**Comment by OpticFusion1** [comment]
> > Aurora 4.1 and https://www.spigotmc.org/resources/lightapi.4510/ 5.1.0
> 
> Paper ?

Spigot. It should work on BOTH however

**Comment by BeYkeRYkt** [comment]
> Spigot. It should work on BOTH however

I succeeded in repeating the problem by updating the JDK to version 17 and rebuilding Aurora with the new JDK. I also noticed that you include LightAPI in the final build, which causes problems, since LightAPI was not designed for such use. LightAPI must work in a standalone version for proper initialization of the library. The patch fix is presented [here](https://github.com/BeYkeRYkt/Aurora/commit/06d2334593779c5e5906b3d95627783be04cb585).
